public/app/types/unified-alerting.ts

```diff
@@ -149,6 +150,20 @@ export interface RuleWithLocation<T = RulerRuleDTO> {
   rule: T;
 }
 
+export const GrafanaRulesSourceSymbol = Symbol('grafana');
```

Comment:
Why have we gone with a symbol for this now? As we have `GRAFANA_RULES_SOURCE_NAME` all over the code already, and I feel this is another way to do the same thing? 🤔

Reply:
With this big refactoring I felt it's a good time to improve our typings and interfaces
The idea is that this symbol should phase out `GRAFANA_RULES_SOURCE_NAME`

There are some issues with `GRAFANA_RULES_SOURCE_NAME`
- It's just a string
- It's difficult to provide unified identifier interface for GMA and DS data sources
- There is nothing preventing your from creating a data source of Id `grafana`

This symbol mainly exists for the interface below
```
export interface ExternalRulesSourceIdentifier {
  uid: string;
  name: string;
}
export interface GrafanaRulesSourceIdentifier {
  uid: typeof GrafanaRulesSourceSymbol;
  name: typeof GRAFANA_RULES_SOURCE_NAME;
}
```
We wanted to achieve a unified rule source identifier for both GMA and DS and phase out `RulesSource` which is DS settings or 'grafana'

Reply:
There's probably still a place for `GRAFANA_RULES_SOURCE_NAME` wherever we actually need to identify a Grafana rules source by string (ie. when the URL contains something like `?alertmanager=grafana`).

We could also use the Symbol's description 

```
const s = Symbol('grafana'); 
console.log(s.description); // grafana
``` 

but that feels a bit awkward to me since we can choose to update the description (and should be something that is meaningful) but we want URLs to be stable.